modules/gui/src/com/haulmont/cuba/gui/xml/layout/loaders/FormLoader.java

```diff
@@ -171,9 +184,13 @@ protected Component loadComponent(Element element, @Nullable String columnWidth)
 
         // Set default width
         String componentWidth = element.attributeValue("width");
-        if (Strings.isNullOrEmpty(componentWidth)
-                && columnWidth != null) {
-            component.setWidth(columnWidth);
+        if (Strings.isNullOrEmpty(componentWidth)) {
+            if (columnWidth != null) {
+                component.setWidth(columnWidth);
+            }
+            if (flex != null) {
```

Comment:
It seems that `columnWidth` should take precedence, but in your implementation, `flex` has precedence. Also, there is a case when `component.setWidth()` is called twice.

Reply:
My initial thought was to make `flex` to take precedence, but I agree it's better the other way around.. and I adjusted the code to set the width only once.